src/main/java/com/saucedemo/pages/LoginPage.java

```diff
@@ -47,16 +43,11 @@ public LoginPage enterLoginFields(User user) {
 
 
     public LoginPage enterUsername(String username) {
-        // TODO remove commented-out code if it is not needed or make initialisation of WebDriverWait in the constructor
-//        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));
-//        wait.until(ExpectedConditions.visibilityOf(usernameArea));
-
         if(username != null) {
             usernameArea.sendKeys(username);
             logger.debug("Entered username: {}", username);
         } else {
             String password = passwordArea.getAttribute("value");;
-            // TODO I'm not quite understand why do we need get password and refresh page if username is empty
```

Comment:
I assume page is doing chashing (or something like that), so whithout refresh input fields remember old values which was set to them even after using method clear, so in the button click moment the old values are send.